gdx/src/com/badlogic/gdx/graphics/Color.java

```diff
@@ -358,10 +401,10 @@ public static Color valueOf (String hex) {
 	 * @see #toString() */
 	public static Color valueOf (String hex, Color color) {
 		hex = hex.charAt(0) == '#' ? hex.substring(1) : hex;
-		color.r = Integer.parseInt(hex.substring(0, 2), 16) / 255f;
-		color.g = Integer.parseInt(hex.substring(2, 4), 16) / 255f;
-		color.b = Integer.parseInt(hex.substring(4, 6), 16) / 255f;
-		color.a = hex.length() != 8 ? 1 : Integer.parseInt(hex.substring(6, 8), 16) / 255f;
+		color.setR(Integer.parseInt(hex.substring(0, 2), 16) / 255f);
```

Comment:
This and all Color-modifying static methods after this will be broken if given an `ImmutableColorCpy`, since the given object will create a modified copy and discard it once per channel. Then it will return the original, unmodified `ImmutableColorCpy`.

Reply:
This is true, and its one reason why I prefer we don't do the `ImmutableColorCpy` route, and instead go the exception route.